src/ex_04/myrand.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Kaspar Lilja'
+__email__ = 'kalilja@nmbu.no'
+
+
+class LCGRand:
+
+    def __init__(self, seed):
+        self.seed = seed
+
+    def rand(self):
+        a = 16807
+        m = 2 ** 31 - 1
+        self.seed = (a * self.seed) % m
+        return self.seed
+
+
+class ListRand:
+
+    def __init__(self, randomlist):
+        self.randomlist = randomlist
+        self.position = 0
+
+    def rand(self):
+        if self.position >= len(self.randomlist):
+            raise RuntimeError
+        else:
+            temp_pos = self.position
+            self.position += 1
+            return self.randomlist[temp_pos]
```

Comment:
Missing docstrings, also the `ListRand` could me more idiomatic python. See example:
```python
class ListRand:

    def __init__(self, rand_list):
        """Returns numbers given drawn from the end of provided list

        Parameters
        ----------
        rand_list : list
            list of random numbers
        """
        self.rand_list = rand_list.copy()

    def rand(self):
        if len(self.rand_list) > 0:
            return self.rand_list.pop()
        raise RuntimeError
```

Reply:
admittedly my solution doesn't keep the list intact within the class. But lets say that's by design.